### PR TITLE
Fix uninitialized variables found by address santizer

### PR DESCRIPTION
**Pull request by DeadParrot**
These fixes could cause some diffs -- let's see what the CI testing reveals.

### CODE REVIEW

**Comment by Myoldmopar** [comment]
It looks like no diffs here.  It does affect construction of the `HVACNodeConvergLogStruct` structure, but it doesn't seem to cause a problem.

I'm inclined to merge this in now.  @mjwitte @EnergyArchmage does anything pop out to you with these changes?

**Comment by DeadParrot** [comment]
@Myoldmopar these are actual bugs: those uninitialized variables were being used with non-zero/false values in at least some builds. If fixing these causes a problem then we have other bugs to find.

File: src/EnergyPlus/DataConvergParams.hh [hunk 1]
```diff
@@ -101,6 +101,10 @@ namespace DataConvergParams {
 
 		// Default Constructor
 		HVACNodeConvergLogStruct() :
+			NodeNum( 0 ),
+			NotConvergedHumRate( false ),
```
Comment (mjwitte) [inline src/EnergyPlus/DataConvergParams.hh]: @DeadParrot These three NotConverge\* variables never seem to be used.  @EnergyArchmage Can we delete them and the loops that are setting them?  Or did I miss something?

Comment (EnergyArchmage) [inline src/EnergyPlus/DataConvergParams.hh]: Yes I guess they can be deleted.  I don't recall the details as to why they are there but I agree they can be deleted.

**Comment by DeadParrot** [comment]
@mjwitte I'm fine with you or @EnergyArchmage doing the "pruning" on this branch to avoid needing a separate PR. Or confirm that it is NotConvergedHumRate, NotConvergedMassFlow, and NotConvergedTemp in these diffs and I can do it.

**Comment by mjwitte** [comment]
@DeadParrot In [DataConvergParams.hh HVACNodeConvergLogStruct](https://github.com/NREL/EnergyPlus/blob/0ed7c081fb5746854702f405aa632f800e46465a/src/EnergyPlus/DataConvergParams.hh#L95-L97), you can delete NotConvergedHumRate, NotConvergedMassFlow, and NotConvergedTemp.  

They are assigned here in [HVACManager::SimHVAC](https://github.com/NREL/EnergyPlus/blob/0ed7c081fb5746854702f405aa632f800e46465a/src/EnergyPlus/HVACManager.cc#L969-L972) and at various points in the same for loop.  Those lines can be removed.

**Comment by DeadParrot** [comment]
@mjwitte OK, done.

**Comment by DeadParrot** [comment]
I merged in develop but there are still CI fails due to fewer variables. No idea...

**Comment by mjwitte** [comment]
Looks all good now.

**Comment by mjwitte** [comment]
@Myoldmopar This looks ready to drop in.

**Pull request merged**